mne_connectivity/spectral/time.py

```diff
@@ -166,48 +205,71 @@ def spectral_connectivity_time(data, names=None, method='coh', indices=None,
     n_pairs = len(source_idx)
 
     # frequency checking
-    if freqs is not None:
+    if cwt_freqs is not None:
         # check for single frequency
-        if isinstance(freqs, (int, float)):
-            freqs = [freqs]
+        if isinstance(cwt_freqs, (int, float)):
+            cwt_freqs = [cwt_freqs]
         # array conversion
-        freqs = np.asarray(freqs)
+        cwt_freqs = np.asarray(cwt_freqs)
         # check order for multiple frequencies
-        if len(freqs) >= 2:
-            delta_f = np.diff(freqs)
+        if len(cwt_freqs) >= 2:
+            delta_f = np.diff(cwt_freqs)
             increase = np.all(delta_f > 0)
             assert increase, "Frequencies should be in increasing order"
 
-        # frequency mean
-        if foi is None:
-            foi_idx = foi_s = foi_e = None
-            f_vec = freqs
-        else:
-            _f = xr.DataArray(np.arange(len(freqs)), dims=('freqs',),
-                              coords=(freqs,))
-            foi_s = _f.sel(freqs=foi[:, 0], method='nearest').data
-            foi_e = _f.sel(freqs=foi[:, 1], method='nearest').data
-            foi_idx = np.c_[foi_s, foi_e]
-            f_vec = freqs[foi_idx].mean(1)
+    # compute frequencies to analyze based on number of samples,
+    # sampling rate, specified wavelet frequencies and mode
+    freqs = _compute_freqs(n_times, sfreq, cwt_freqs, mode)
```

Comment:
What if `cwt_freqs` is `None`? Will  that not throw an error?

Reply:
`_compute_freqs` will throw an error if `mode='cwt_morlet` and `cwt_freqs is None`. I think this should be the correct behaviour.

Reply:
Also fine, but again should be clearer from the documentation - or split into two functions - this will also be quite difficult to maintain.

Reply:
I will try to improve the documentation for now.

Reply:
I added some clarification to the docstring.